testing/citest/tests/azure_smoke_test.py

```diff
@@ -0,0 +1,303 @@
+    "--azure_client_id=$AZURE_CLIENT_ID", \
+    "--azure_app_key=$AZURE_APP_KEY", \
+    "--test_azure_rg_location=$TEST_AZURE_RG_LOCATION"
+
+"""
+# Standard python modules.
+import sys
+
+# citest modules.
+import citest.azure_testing as az
+import citest.json_contract as jc
+import citest.json_predicate as jp
+import citest.service_testing as st
+import citest.base
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+
+class AzureSmokeTestScenario(sk.SpinnakerTestScenario):
+  """Defines the scenario for the smoke test.
+
+  This scenario defines the different test operations.
+  We're going to:
+    Create a Spinnaker Application
+    Create a Spinnaker Security Group
+    Delete each of the above (in reverse order)
+  """
+
+  @classmethod
+  def new_agent(cls, bindings):
+    """Implements citest.service_testing.AgentTestScenario.new_agent."""
+    agent = gate.new_agent(bindings)
+    agent.default_max_wait_secs = 180
+    return agent
+
+  def __init__(self, bindings, agent=None):
+    """Constructor.
+    Args:
+      bindings: [dict] The data bindings to use to configure the scenario.
+      agent: [GateAgent] The agent for invoking the test operations on Gate.
+    """
+    super(AzureSmokeTestScenario, self).__init__(bindings, agent)
+    bindings = self.bindings
+
+    # create default sec grp name  
+    # Spinnaker-azure appends region to actual RG used by an app to support multi-region
+    # pylint: disable=invalid-name
+    self.TEST_SECURITY_GROUP = 'sec_grp_' + self.bindings['TEST_APP']
+    self.TEST_SECURITY_GROUP_RULE_1=self.TEST_SECURITY_GROUP + '_RULE_1'
+    self.TEST_SECURITY_GROUP_RG = self.bindings['TEST_APP'] + '-' + self.bindings['TEST_AZURE_RG_LOCATION']
+
+  #TODO: Perform the login to Azure withTEST_APP the SPN passed as parameters
+
+  def create_app(self):
+    """Creates OperationContract that creates a new Spinnaker Application and validate its creation in Azure Storage.
+    """
+    email = self.bindings.get('TEST_EMAIL', 'testuser@testhost.org')
+    payload = self.agent.make_json_payload_from_kwargs(
+            job=[{
+                'type': 'createApplication',
+                'account': self.bindings['SPINNAKER_AZURE_ACCOUNT'],
+                'application': {
+                    'name': self.bindings['TEST_APP'],
+                    'description': 'Gate Testing Application for Azure',
+                    'email': email
+                },
+                'user': '[anonymous]'
+            }],
+            description= 'Test - create application {app}'.format(app=self.bindings['TEST_APP']),
+            application=self.bindings['TEST_APP'])
+
+    builder = az.AzContractBuilder(self.az_observer)
+    (builder.new_clause_builder(
+        'Application Created', retryable_for_secs=30)
+      .collect_resources(
+          az_resource='storage',
+          command='blob',
+          args=['exists', '--container-name', 'front50',
```

Comment:
What is "front50" here? Is this the name of an azure resource (e.g. are you assuming spinnaker is configured with a container named "front50" ? Is this something you hardcode into spinnaker, or was this a deployment choice? If a deployment choice, it should be parameterized (e.g. get it from the agent.deployed_config) so you can run the test against any deployment without having to overprescribe configuration.